### PR TITLE
Add sw-precache

### PR DESCRIPTION
**Pull request by denar90**
Related to #469

### CODE REVIEW

**Comment by denar90** [comment]
This PR is ready for review. Before merging it we need to solve #454 (moving on https).
P.S. For reviewer. Please squash my commits 😄

**Comment by denar90** [comment]
Closing this until we can find solution for using HTTPS.

**Comment by denar90** [comment]
Reopening that due to #469. I'll resolve and retest stuff later...

**Comment by paulfalgout** [comment]
For this we'll want to do something to redirect people to https:// right?  Like currently either http:// or https:// works.

**Comment by denar90** [comment]
@paulfalgout 👍

**Comment by denar90** [comment]
I've updated stuff. Also added some improvements. 

---

So right now we have 88% score for [Lighthouse](https://github.com/GoogleChrome/lighthouse/)
Using prod with `https` will higher this value up.

image

---

To reach 100% we have to solve - https://github.com/marionettejs/marionettejs.com/issues/495

**Comment by denar90** [comment]
Since @thejameskyle added redirect to `https`, this PR ready for review :)

**Comment by samccone** [comment]
💥 +1

File: src/sections/_nav.jade [hunk 1]
```diff
@@ -1,4 +1,4 @@
-input.hidden-visually(type="checkbox", id="toggle-search")
+input.hidden-visually(type="checkbox", id="toggle-search", hidden)
```
Comment (denar90) [inline src/sections/_nav.jade]: @peterblazejewicz yeap you are right 👍  I have to recheck if this stuff https://github.com/marionettejs/marionettejs.com/blob/master/src/js/search.js#L6 will work though.

**Pull request merged**